### PR TITLE
[Google Threat Intelligence] Add curated collections commands and improve polling commands

### PR DESCRIPTION
**Pull request by pabloperezj**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligence/GoogleThreatIntelligence.py [hunk 1]
```diff
@@ -37,6 +37,15 @@
     'VERDICT_MALICIOUS': 'MALICIOUS',
 }
 
+TYPE_TO_ENDPOINT = {
+    'file': 'files',
+    'hash': 'files',
+    'domain': 'domains',
+    'url': 'urls',
+    'ip': 'ip_addresses',
+    'ip_address': 'ip_addresses',
```
Comment (israelpoli) [inline Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligence/GoogleThreatIntelligence.py:46]:
```suggestion
```
need it? I don't see this key being used

File: Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligence/GoogleThreatIntelligence.py [hunk 2]
```diff
@@ -2631,6 +2676,68 @@ def get_assessment_command(client: Client, score_calculator: ScoreCalculator, ar
...
+            'link': f'https://www.virustotal.com/gui/collection/{collection["id"]}'
+        })
+
+    type_str = collection_type.replace('-', ' ')
+    type_context = type_str.title().replace(' ', '')
+    type_title = f'{type_str[:-1]}ies' if type_str.endswith('y') else f'{type_str}s'
+
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.{type_context}',
+        'id',
```
Comment (israelpoli) [inline Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligence/GoogleThreatIntelligence.py:2705]:
```suggestion
        outputs_prefix=f'{INTEGRATION_ENTRY_CONTEXT}.{type_context}',
        outputs_key_field='id',
```
Clearer

File: Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligence/GoogleThreatIntelligence.py [hunk 3]
```diff
@@ -2631,6 +2676,68 @@ def get_assessment_command(client: Client, score_calculator: ScoreCalculator, ar
     )
 
 
+def _get_curated_collections_command(client: Client, args: dict, collection_type: str) -> CommandResults:
+    """Get Google Threat Intelligence collections for a given resource."""
+    resource = args['resource']
+    resource_type = args.get('resource_type', 'file').lower()
+    if resource_type not in ('file', 'hash', 'ip', 'url', 'domain'):
+        raise DemistoException(f'Could not find resource type of "{resource_type}"')
```
Comment (israelpoli) [inline Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligence/GoogleThreatIntelligence.py:2684]:
```suggestion
```
I see that the same check is done inside the curated_collections method

**Review by israelpoli — COMMENTED** [review]
Hi @pabloperezj thanks for your contribution
It looks great
see my comments below

And if you could share a demo video of the 3 new commands to see that it works as expected, or alternatively, a demo meeting will be scheduled next week

thanks

**Comment by pabloperezj** [comment]
> Hi @pabloperezj thanks for your contribution It looks great see my comments below
> 
> And if you could share a demo video of the 3 new commands to see that it works as expected, or alternatively, a demo meeting will be scheduled next week
> 
> thanks

Hi @israelpoli,

I have just uploaded the demo video to the virustotal channel on Slack :)

Thanks!

**Comment by israelpoli** [comment]
@pabloperezj looks good, I will take it to the next step

There is still no documentation for the commands in the integration's README file, but I will generate it using the `demisto-sdk` command.

thanks

**Pull request merged**